platform/inprocess/mod.rs

```diff
@@ -0,0 +1,359 @@
+// Copyright 2015 The Servo Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::sync::mpsc;
+use std::sync::{Arc, Mutex, Condvar};
+use std::collections::hash_map::HashMap;
+use std::cell::{RefCell};
+use std::slice;
+use std::fmt::{self, Debug, Formatter};
+use std::cmp::{PartialEq};
+use std::ops::Deref;
+use std::mem;
+
+use uuid::Uuid;
+
+struct SenderRecord {
+    sender: MpscSender,
+    lock: Mutex<bool>,
+    condvar: Condvar,
+}
+
+impl SenderRecord {
+    fn new(sender: MpscSender) -> SenderRecord {
+        SenderRecord {
+            sender: sender,
+            lock: Mutex::new(false),
+            condvar: Condvar::new(),
+        }
+    }
+
+    fn wait_for_acquire(&self) -> () {
+        let mut started = self.lock.lock().unwrap();
+        while !*started {
+            started = self.condvar.wait(started).unwrap();
+        }
+    }
+
+    fn signal_acquisition(&self) -> () {
+        let mut notified = self.lock.lock().unwrap();
+        *notified = true;
```

Comment:
In theory I'd need the sender to also block until the receiver actually finishes receiving.  But maybe that's not actually necessary in this case, since the actual connection setup can't really fail...